htlcswitch/link.go

```diff
@@ -2696,6 +2757,138 @@ func (l *channelLink) getDustClosure() dustClosure {
 	return dustHelper(chanType, localDustLimit, remoteDustLimit)
 }
 
+// getCommitFee returns either the local or remote CommitFee in satoshis. This
+// is used so that the Switch can have access to the commitment fee without
+// needing to have a *LightningChannel.
+//
+// NOTE: Part of the dustHandler interface.
+func (l *channelLink) getCommitFee(remote bool) btcutil.Amount {
+	if remote {
+		return l.channel.State().RemoteCommitment.CommitFee
+	}
+
+	return l.channel.State().LocalCommitment.CommitFee
+}
+
+// isDustWithFee returns whether or not the new proposed fee-rate increases the
+// total dust and fees within the channel past the configured dust threshold.
+// It first calculates the dust sum over every update in the update log with
+// the proposed fee-rate and taking into account both the local and remote dust
+// limits. It uses every update in the update log instead of what is actually
+// on the local and remote commitments because it is assumed that in a
+// worst-case scenario, every update in the update log could theoretically be
+// on either commitment transaction and this needs to be accounted for with
+// this fee-rate. It then calculates the local and remote commitment fees given
+// the proposed fee-rate. Finally, it tallies the results and determines if the
+// dust threshold has been exceeded.
+func (l *channelLink) isDustWithFee(feePerKw chainfee.SatPerKWeight) (bool,
```

Comment:
Yeah I think redefining this to just be `exceedsFeeExposureLimit` or something to that effect would make more sense.